operators/src/source/gdal_source.rs

```diff
@@ -445,75 +287,65 @@ where
                     GridBoundingBox2D::new(tile_idx_ul, tile_idx_lr)?,
                 )?;
 
-                let mut tile_raster = Grid2D::new_filled(tile_grid, T::zero(), None);
+                let mut tile_raster = Grid2D::new_filled(output_shape, T::zero(), None);
                 tile_raster.grid_blit_from(dataset_raster)?;
                 tile_raster
             }
         };
 
-        Ok(RasterTile2D::new_with_tile_info(
-            time_interval,
-            tile_information,
-            result_raster,
-        ))
+        Ok(result_raster)
     }
 
     ///
     /// A stream of `RasterTile2D`
     ///
     pub fn tile_stream(
         &self,
-        bbox: BoundingBox2D,
-        time_interval: TimeInterval,
-        spatial_resolution: SpatialResolution,
+        query: QueryRectangle,
+        info: GdalLoadingInfoPart,
     ) -> BoxStream<Result<RasterTile2D<T>>> {
+        let spatial_resolution = query.spatial_resolution;
+        let geo_transform = info.params.geo_transform;
+
         // adjust the spatial resolution to the sign of the geotransform
-        let x_signed = if self
-            .dataset_information
-            .geo_transform()
-            .x_pixel_size
-            .is_sign_positive()
+        let x_signed = if geo_transform.x_pixel_size.is_sign_positive()
             && spatial_resolution.x.is_sign_positive()
         {
             spatial_resolution.x
         } else {
             spatial_resolution.x * -1.0
         };
 
-        let y_signed = if self
-            .dataset_information
-            .geo_transform()
-            .y_pixel_size
-            .is_sign_positive()
+        let y_signed = if geo_transform.y_pixel_size.is_sign_positive()
             && spatial_resolution.y.is_sign_positive()
         {
             spatial_resolution.y
         } else {
             spatial_resolution.y * -1.0
         };
 
-        let tiling_strategy =
-            TilingStrategy::new_with_tiling_spec(self.tiling_specification, x_signed, y_signed);
+        let tiling_strategy = self.tiling_specification.strategy(x_signed, y_signed);
 
-        stream::iter(self.time_tile_iter(tiling_strategy, time_interval, bbox))
-            .map(move |(time, tile)| {
-                (
-                    self.gdal_params.clone(),
-                    self.dataset_information.clone(),
-                    time,
+        // TODO: avoid cloning loading_info?
+        stream::iter(tiling_strategy.tile_information_iterator(query.bbox))
+            .map(move |tile| {
+                Ok(RasterTile2D::new_with_tile_info(
+                    info.time,
                     tile,
-                )
-            })
-            .then(|(gdal_params, dataset_information, time, tile)| {
-                Self::load_tile_data_async(gdal_params, dataset_information, time, tile)
+                    Self::load_tile_data(
```

Comment:
why not use the async version?

Reply:
4c1c5e5ab8f123cf7c8db87f80ed7d381a525ca8